politeiad/api/v1/v1.go

```diff
@@ -32,12 +33,23 @@ const (
 
 	ChallengeSize = 32 // Size of challenge token in bytes
 
-	// Status codes (set and get)
-	StatusInvalid     StatusT = 0 // Invalid status
-	StatusNotFound    StatusT = 1 // Proposal not found
-	StatusNotReviewed StatusT = 2 // Proposal has not been reviewed
-	StatusCensored    StatusT = 3 // Proposal has been censored
-	StatusPublic      StatusT = 4 // Proposal is publicly visible
+	// Status codes
+	StatusInvalid                     StatusT = 0
+	StatusInvalidRequestPayload       StatusT = 1
+	StatusInvalidChallenge            StatusT = 2
+	StatusInvalidProposalName         StatusT = 3
+	StatusInvalidFileDigest           StatusT = 4
+	StatusInvalidBase64               StatusT = 5
+	StatusInvalidMIMEType             StatusT = 6
+	StatusUnsupportedMIMEType         StatusT = 7
+	StatusInvalidPropStatusTransition StatusT = 8
```

Comment:
We need to be consistent with naming and we have not been.  The following status defines start with prop and that is backwards.  All Statuses should start with Status followed by the descriptor OR the other way around.  I find descriptorType a bit harder to read but it groups things nicely.  So like: ErrorStatus... PropStatus... CommentStatus.

This really bothered me the other day when writing code and seeing that we are all over the place with this.  We should pick one and stick with it.  And we need to go back over the API and correct the backwards ones.  We are getting to a point where the API is going to become solidified so we want to do this soon.

Reply:
This goes for code as well btw.  Functions should start with the thing they belong to like: propChangeThis and commentNew etc etc.

Reply:
Changed Status -> ErrorStatus.